check50/c.py

```diff
@@ -60,8 +60,9 @@ def compile(*files, exe_name=None, cc=CC, **cflags):
     # Strip out ANSI codes
     stdout = re.sub(r"\x1B\[[0-?]*[ -/]*[@-~]", "",  process.stdout())
 
+    # Log the last max_log_lines lines of output in case compilation fails
     if process.exitcode != 0:
-        for line in stdout.splitlines():
+        for line in stdout.splitlines()[-max_log_lines:]:
```

Comment:
Hmm, does it make more sense to show the first n lines instead of the last n? Should we limit the number of chars (instead of the number of lines) to control the size more?

Reply:
Should we consider using `-ferror-limit` too? https://clang.llvm.org/docs/UsersManual.html#cmdoption-ferror-limit

Reply:
I think the last n lines make more sense, as that more likely shows what caused the check to fail and the information just before it is probably more relevant. Imagine seeing a traceback from Python perhaps, and not the actual error in the end ;)

`-ferror-limit` wouldn't help, because it's the warnings that blow up the results. Just checked again, and there are indeed just 20 errors shown (default value of `-ferror-limit`), but a good 629 warnings. Sadly I can't find `-fwarning-limit` or anything of the sorts for clang, but perhaps I overlooked something.

Reply:
I wasn't thinking about Python. For C, I think the first few lines are more relevant for sure, but I can see why it'd be useful to see the last few lines with Python. As a middle ground, should we just show 

```
the first n/2 lines 
... 
the last n/2 lines
```

?

Reply:
Deal